serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt

```diff
@@ -51,19 +51,24 @@ class MatrixLogin(val login_message: String, val mclient: MatrixClient): MatrixS
     }
 }
 class MatrixRooms(val msession: MatrixSession): MatrixState() {
-    fun test(): MatrixState {
-        return MatrixLogin("${msession.test()}, now going back to login for now\n", MatrixClient())
+    fun getRoom(): MatrixState {
+        return MatrixChatRoom(msession)
+    }
+}
+class MatrixChatRoom(val msession: MatrixSession): MatrixState() {
+    fun sendMessage(msg : String): MatrixState {
+        return MatrixLogin("${msession.test(msg)}, now going back to login for now\n", MatrixClient())
     }
 }
 class MatrixSession(val client: HttpClient, val access_token: String) {
-    fun test(): String {
+    fun test(msg : String): String {
 
         val result = runBlocking {
 
             val room_id = "!bwqkmRobBXpTSDiGIw:synapse.room409.xyz"
             val message_confirmation = client.put<EventIdResponse>("https://synapse.room409.xyz/_matrix/client/r0/rooms/$room_id/send/m.room.message/23?access_token=$access_token") {
```

Comment:
I think we might want to handle failure to send here, either because of loss of internet or being rate limited *cough cough Enexone* Perhaps with another optional message, this time in MatrixChatRoom?

Reply:
Agreed, I'll add some rror handling.